File: php/28-implement-strstr.php
<?php
class Solution {

    /**
     * @param String $haystack
     * @param String $needle
     * @return Integer
     */
    function strStr($haystack, $needle) {
        $lenh = strlen($haystack);
        $lenn = strlen($needle);
        if ($lenn == 0) {
            return 0;
        }
        if ($lenh < $lenn) {
            return -1;
        }
        $h = 0;
        while ($h < $lenh) {
            if ($needle[0] == $haystack[$h] && substr($haystack, $h, $lenn) == $needle) {
                return $h;
            }
            $h++;
        }
        return -1;
    }
}

File: go/28-implement-strstr.go
package main

func strStr(haystack string, needle string) int {

	hs := []byte(haystack)
	ns := []byte(needle)
	if len(ns) == 0 {
		return 0
	}
	if len(ns) > len(hs) {
		return -1
	}
	for k, v := range hs {
		if v == ns[0] {
			allMatch := true
			ni := 1
			for i := 1; i < len(ns); i++ {
				if i+k >= len(hs) || hs[i+k] != ns[ni] {
					allMatch = false
					break
				}
				ni++
			}
			if allMatch {
				return k
			}
		}
	}
	return -1
}
